From 6e85fa3570460d11ad40060b3833f9d554143c34 Mon Sep 17 00:00:00 2001
From: Mridul Singh <mridulsingh@codeaurora.org>
Date: Wed, 1 Mar 2017 11:37:34 -0800
Subject: [PATCH 48/70] Camera: Add NULL check on mDevice to avoid crash.

NULL check on mDevice before calling cancelAutoFocus.
Also adding the same check in disconnect.
---
 services/camera/libcameraservice/api1/Camera2Client.cpp | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/services/camera/libcameraservice/api1/Camera2Client.cpp b/services/camera/libcameraservice/api1/Camera2Client.cpp
index 65faac94b..2095e0c4e 100644
--- a/services/camera/libcameraservice/api1/Camera2Client.cpp
+++ b/services/camera/libcameraservice/api1/Camera2Client.cpp
@@ -453,9 +453,10 @@ binder::Status Camera2Client::disconnect() {
 
     ALOGV("Camera %d: Disconnecting device", mCameraId);
 
-    mDevice->disconnect();
-
-    mDevice.clear();
+    if (mDevice != nullptr) {
+        mDevice->disconnect();
+        mDevice.clear();
+    }
 
     CameraService::Client::disconnect();
 
@@ -935,6 +936,7 @@ void Camera2Client::stopPreview() {
     Mutex::Autolock icl(mBinderSerializationLock);
     status_t res;
     if ( (res = checkPid(__FUNCTION__) ) != OK) return;
+    if (mDevice == 0) return;
     stopPreviewL();
 }
 
@@ -1400,7 +1402,7 @@ status_t Camera2Client::cancelAutoFocus() {
     ALOGV("%s: Camera %d", __FUNCTION__, mCameraId);
     status_t res;
     if ( (res = checkPid(__FUNCTION__) ) != OK) return res;
-
+    if (mDevice == 0) return NO_INIT;
     int triggerId;
     {
         SharedParameters::Lock l(mParameters);
-- 
2.11.0

